class/practice5.2/MegaData.h

```diff
@@ -0,0 +1,42 @@
+#ifndef MEGADATA_H
+#define MEGADATA_H
+
+#include <vector>
+#include <memory>
+#include <mutex>
+
+class MegaData {
+public:
+    std::vector<float> smallArray;
+    std::vector<double> bigArray;
+
+    MegaData();
+    void reset();
+};
+
+class MegaDataPool {
+private:
+    std::vector<std::unique_ptr<MegaData>> pool;
+    std::vector<MegaData*> available;
+    mutable std::mutex poolMutex;
+
+    MegaDataPool(size_t size);
+
+    MegaDataPool(const MegaDataPool&) = delete;
+    MegaDataPool& operator=(const MegaDataPool&) = delete;
```

Comment:
Minor: We can also follow [the rule of 5 here](https://www.geeksforgeeks.org/rule-of-five-in-cpp/) e.g. delete move constructor as well.